### PR TITLE
CHET-89: Add OS/FS/DB analytics and add some enumerations to whitelist.

### PR DESCRIPTION
**Pull request by tarka**

### CODE REVIEW

File: core/src/main/java/com/atlassian/migration/datacenter/core/util/AnalyticsHelper.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
+package com.atlassian.migration.datacenter.core.util;
+
+import org.apache.commons.lang3.SystemUtils;
+
+public class AnalyticsHelper
```
Comment (vicav) [inline core/src/main/java/com/atlassian/migration/datacenter/core/util/AnalyticsHelper.java:5]: /nit: Not sure if AnalyticsHelper wrapper around OsType enum is needed. I'd expect simply the OsType enum with the method getSystemOsType()

File: core/src/main/java/com/atlassian/migration/datacenter/core/aws/AWSMigrationService.java [hunk 2]
```diff
@@ -117,12 +119,12 @@ public synchronized void transition(MigrationStage to) throws InvalidMigrationSt
 
         if (!currentStage.isValidTransition(to)) {
             eventPublisher.publish(new MigrationTransitionFailedEvent(applicationConfiguration.getPluginVersion(),
-                                                                      currentStage.toString(), to.toString()));
+                                                                      currentStage, to));
```
Comment (theghostwhoforks) [inline core/src/main/java/com/atlassian/migration/datacenter/core/aws/AWSMigrationService.java:122]: I'm wondering if we should push this into the `transition` method within the statemachine? That way the publish logic will be controlled as a state transitions without the services calling the transition having to explicitly invoke the publish method?

or worst case, we just call transition with an optional/nullable `eventPublisher` callback

File: core/src/main/java/com/atlassian/migration/datacenter/core/util/AnalyticsHelper.java (hunk 1 above)
Comment (tarka) [reply core/src/main/java/com/atlassian/migration/datacenter/core/util/AnalyticsHelper.java]: Good point, done.

File: core/src/main/java/com/atlassian/migration/datacenter/core/aws/AWSMigrationService.java (hunk 2 above)
Comment (tarka) [reply core/src/main/java/com/atlassian/migration/datacenter/core/aws/AWSMigrationService.java:122]: Unfortunately the AWSMigrationService is the closest thing we have to a state machine currently.

**Pull request merged**